notebooks/colab_metl_predicting.ipynb

```diff
@@ -0,0 +1,2036 @@
+      "execution_count": null,
+      "metadata": {
+        "id": "ecLAU9IOrLGK",
+        "cellView": "form"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Transform input from 1 based indexing to zero based indexing\n",
+        "# @markdown Select indexing for residue mutations\n",
+        "indexing = \"0\" # @param ['0', '1']"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ixhZbYm8rLGM"
+      },
+      "source": [
+        "Since both file and string variants give the same result, we only need to use one moving forwards. We will use the string_variants variable.\n",
+        "\n",
+        "To predict with METL, we will need to use the loaded model and encoder with our variables we defined above. We will wrap this in a for loop to predict on all of our variants as we have multiple lines of them."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "th1JyojWrLGN",
+        "cellView": "form"
+      },
+      "outputs": [],
+      "source": [
+        "# @title METL Predicting loop\n",
+        "output = []\n",
+        "\n",
+        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
+        "\n",
+        "metl = metl.to(device)\n",
+        "\n",
+        "if indexing == \"1\":\n",
+        "  predict_variants = to_zero_based(variants)\n",
+        "else:\n",
+        "  predict_variants = variants\n",
+        "\n",
+        "for variant in predict_variants:\n",
+        "    # First in METL we need to encode our variants\n",
+        "    if not isinstance(variant, list):\n",
+        "      variant = json.loads(variant)\n",
+        "    encoded_variants = metl.encoder.encode_variants(wildtype, variant)\n",
+        "\n",
+        "    #Next, we predict\n",
+        "    with torch.no_grad():\n",
+        "        if pdb_file_path:\n",
+        "            predictions = metl(torch.tensor(encoded_variants).to(device), pdb_fn=pdb_file_path)\n",
+        "        else:\n",
+        "            predictions = metl(torch.tensor(encoded_variants).to(device))\n",
+        "\n",
+        "        output.append({\n",
+        "            \"wt\": wildtype,\n",
+        "            \"variants\": variant,\n",
+        "            \"logits\": predictions.tolist()\n",
+        "        })"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "pO8jdBssrLGO"
+      },
+      "source": [
+        "Finally, we will save our output. We will save our output as a list of JSON Objects."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "nYz6NRB_rLGP",
+        "cellView": "form"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Saving the predictions\n",
+        "with open('./output.json', 'w') as f:\n",
+        "    f.write(json.dumps(output, indent=2))"
```

Comment:
Can we also display the predictions?

Reply:
Yea, I was just worried about the metl models that display embeddings taking up too much screen space. But I can display a cut off view if its going to take too many lines. Even with the scrollable limits it's a lot most of the time.

Reply:
Made it so that the output height was limited and it works nicely

Reply:
This looks great